libtenzir/builtins/functions/community_id.cpp

```diff
@@ -0,0 +1,251 @@
+        if (not seeds) {
+          diagnostic::warning("`community_id` got an argument type mismatch")
+            .primary(*args.seed)
+            .note("expected argument `seed` to be of type `int64`")
+            .emit(ctx);
+          return null_series();
+        }
+      }
+      auto b = arrow::StringBuilder{};
+      check(b.Reserve(eval.length()));
+      auto emit_proto_warning = false;
+      auto emit_port_warning = false;
+      auto emit_seed_warning = false;
+      for (auto i = int64_t{0}; i < eval.length(); ++i) {
+        if (src_ips->array->IsNull(i) or dst_ips->array->IsNull(i)
+            or protos->array->IsNull(i)) {
+          check(b.AppendNull());
+          continue;
+        }
+        const auto* src_ip_ptr = src_ips->array->storage()->GetValue(i);
+        const auto* dst_ip_ptr = dst_ips->array->storage()->GetValue(i);
+        auto src_ip = ip::v6(as_bytes<16>(src_ip_ptr));
+        auto dst_ip = ip::v6(as_bytes<16>(dst_ip_ptr));
+        auto proto = protos->array->GetView(i);
+        auto proto_type = port_type::unknown;
+        if (proto == "tcp") {
+          proto_type = port_type::tcp;
+        } else if (proto == "udp") {
+          proto_type = port_type::udp;
+        } else if (proto == "icmp") {
+          proto_type = port_type::icmp;
+        } else if (proto == "icmp6") {
+          proto_type = port_type::icmp6;
+        } else {
+          emit_proto_warning = true;
+          check(b.AppendNull());
+          continue;
+        }
+        auto seed = uint16_t{0};
+        if (seeds and not seeds->array->IsNull(i)) {
+          auto value = seeds->array->GetView(i);
+          if (value < 0 or value > 65'535) {
+            emit_seed_warning = true;
+            check(b.AppendNull());
+            continue;
+          }
+          seed = detail::narrow_cast<uint16_t>(value);
+        }
+        auto have_src_port = src_ports and not src_ports->array->IsNull(i);
+        auto have_dst_port = dst_ports and not dst_ports->array->IsNull(i);
+        if (have_src_port and have_dst_port) {
+          auto src_port = src_ports->array->GetView(i);
+          auto dst_port = dst_ports->array->GetView(i);
+          auto flow
+            = make_flow(src_ip, dst_ip, detail::narrow<uint16_t>(src_port),
+                        detail::narrow<uint16_t>(dst_port), proto_type);
+          check(b.Append(tenzir::community_id::make(flow, seed)));
+        } else if (have_src_port != have_dst_port) {
+            emit_port_warning = true;
+            check(b.AppendNull());
+            continue;
+        } else {
+          check(b.Append(
+            tenzir::community_id::make(src_ip, dst_ip, proto_type, seed)));
+        }
+      }
+      if (emit_seed_warning) {
+        diagnostic::warning("`community_id` got invalid seed")
+          .primary(*args.seed)
+          .note("`seed` exceeded maximum value of 65,535")
+          .hint("use a seed that's less then the maximum value")
+          .emit(ctx);
```

Comment:
```suggestion
        diagnostic::warning("`seed` must be between 0 and 65535")
          .primary(*args.seed)
          .emit(ctx);
```

YMMV, but I prefer this. Overall information content should be about the same (and I added the 0 lowerbound). You can make similar changes below, if you want to.